libbpfgo/libbpfgo.go

```diff
@@ -478,6 +478,67 @@ func (b *BPFMap) Update(key, value interface{}) error {
 	return nil
 }
 
+type BPFMapIterator struct {
+	b *BPFMap
+
+	err error
+
+	keySize int
+
+	prev []byte
+	next []byte
+}
+
+func (b *BPFMap) Iter(keySize int) *BPFMapIterator {
+	return &BPFMapIterator{
+		b:       b,
+		keySize: keySize,
+		prev:    nil,
+		next:    nil,
+	}
+}
+
+func (it *BPFMapIterator) Next() bool {
+	if it.err != nil {
+		return false
+	}
+
+	prevPtr := unsafe.Pointer(nil)
+	if it.next != nil {
+		prevPtr = unsafe.Pointer(&it.next[0])
+	}
+
+	next := make([]byte, it.keySize)
+	nextPtr := unsafe.Pointer(&next[0])
+
+	errC := C.bpf_map_get_next_key(it.b.fd, prevPtr, nextPtr)
+	if errC == -1 {
+		it.prev = it.next
+		it.next = next
```

Comment:
Quoting from the man page (https://man7.org/linux/man-pages/man2/bpf.2.html):

> If key is the last element, -1 is returned and errno is set to ENOENT. 
> Other possible errno values are ENOMEM, EFAULT, EPERM, and EINVAL.

So, is it correct to do the above lines in case errno != ENOENT?

Reply:
I think you're right, we're missing the last iteration as is, let me fix that!